### PR TITLE
copr: Vectorize `instr`.

### PR DESCRIPTION
**Pull request by cireu**
Signed-off-by: Zhu Zihao <all_but_last@163.com>
PCP #5751 

###  What have you changed?

Vectorize `instr`

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by cireu** [comment]
@sre-bot  /run-integration-copr-test copr-test=pr/100

**Comment by cireu** [comment]
@iosmanthus @mapleFU @lonng @breeswish PTAL

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 1]
```diff
@@ -1241,4 +1257,54 @@ mod tests {
             assert_eq!(output, expect_output);
         }
     }
+
+    #[test]
+    fn test_instr() {
+        let cases: Vec<(&str, &str, i64)> = vec![
+            ("a", "abcdefg", 1),
```
Comment (iosmanthus) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: Would you mind adding some invalid utf-8 char to these cases?

Comment (cireu) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: Passing invalid utf-8 string to string maniplate function is undefined behaviour now(?)

https://github.com/tikv/tikv/pull/5946#pullrequestreview-318766329

**Comment by TennyZhuang** [comment]
LGTM

/approve

**Comment by cireu** [comment]
@sre-bot /merge

**Comment by cireu** [comment]
@breeswish please merge it again.

**Comment by mapleFU** [comment]
Sorry for our unstable CI...

**Comment by cireu** [comment]
```
[2019-12-04T13:50:26.913Z] + TIDB_PATH=./bin/tidb-server

[2019-12-04T13:50:26.913Z] + PD_PATH=./bin/pd-server

[2019-12-04T13:50:26.913Z] + UPGRADE_PART=tikv

[2019-12-04T13:50:26.913Z] + NEW_BINARY=./bin/tikv-server

[2019-12-04T13:50:26.913Z] + OLD_BINARY=./bin/tikv-server-old

[2019-12-04T13:50:26.913Z] + GOPATH=/home/jenkins/agent/workspace/tikv_ghpr_integration_compatibility_test/go

[2019-12-04T13:50:26.913Z] + ./test.sh

[2019-12-04T13:50:26.913Z] compatibletest start

[2019-12-04T13:50:26.913Z] /home/jenkins/agent/workspace/tikv_ghpr_integration_compatibility_test/go

[2019-12-04T13:51:48.295Z] time="2019-12-04T21:51:39+08:00" level=warning msg="ping db failed, retry count 0 err Error 9002: TiKV server timeout"

[2019-12-04T13:52:27.012Z] time="2019-12-04T21:52:20+08:00" level=warning msg="ping db failed, retry count 1 err Error 9002: TiKV server timeout"

[2019-12-04T13:53:05.721Z] time="2019-12-04T21:53:01+08:00" level=warning msg="ping db failed, retry count 2 err Error 9002: TiKV server timeout"

[2019-12-04T13:53:44.387Z] time="2019-12-04T21:53:42+08:00" level=warning msg="ping db failed, retry count 3 err Error 9002: TiKV server timeout"

[2019-12-04T13:54:31.014Z] time="2019-12-04T21:54:23+08:00" level=warning msg="ping db failed, retry count 4 err Error 9002: TiKV server timeout"

[2019-12-04T13:55:09.685Z] time="2019-12-04T21:55:04+08:00" level=warning msg="ping db failed, retry count 5 err Error 9002: TiKV server timeout"

[2019-12-04T13:55:48.353Z] time="2019-12-04T21:55:44+08:00" level=warning msg="ping db failed, retry count 6 err Error 9002: TiKV server timeout"

[2019-12-04T13:56:27.022Z] time="2019-12-04T21:56:25+08:00" level=warning msg="ping db failed, retry count 7 err Error 9002: TiKV server timeout"

[2019-12-04T13:57:13.640Z] time="2019-12-04T21:57:06+08:00" level=warning msg="ping db failed, retry count 8 err Error 9002: TiKV server timeout"

[2019-12-04T13:57:52.306Z] time="2019-12-04T21:57:47+08:00" level=warning msg="ping db failed, retry count 9 err Error 9002: TiKV server timeout"

[2019-12-04T13:58:30.969Z] time="2019-12-04T21:58:29+08:00" level=warning msg="ping db failed, retry count 10 err Error 9002: TiKV server timeout"

[2019-12-04T13:59:17.592Z] time="2019-12-04T21:59:10+08:00" level=warning msg="ping db failed, retry count 11 err Error 9002: TiKV server timeout"

[2019-12-04T13:59:56.259Z] time="2019-12-04T21:59:50+08:00" level=warning msg="ping db failed, retry count 12 err Error 9002: TiKV server timeout"

[2019-12-04T14:00:26.535Z] Sending interrupt signal to process

[2019-12-04T14:00:26.535Z] Killing processes
```

CI failed to connect to TiKV server.

**Comment by mapleFU** [comment]
Seems that it's not only `test_server_transfer_leader_safe` error now , cry

**Comment by mapleFU** [comment]
You can spend time on other prs. I has told my friend that there may be some error with `test_server_transfer_leader_safe`. He will have a look at it tomorrow.

**Comment by cireu** [comment]
Thanks a lot! @mapleFU

**Pull request merged**